### PR TITLE
Added policy in distcp-ng Hive copy to force delete physical data on target if its not managed by the target Hive table or partition

### PR DESCRIPTION
**Pull request by abti**

### CODE REVIEW

**Comment by abti** [comment]
@ibuenros Can you please review?

File: gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/HiveCopyEntityHelper.java [hunk 1]
```diff
@@ -90,6 +90,11 @@
   public static final String EXISTING_ENTITY_POLICY_KEY =
       HiveDatasetFinder.HIVE_DATASET_PREFIX + ".existing.entity.conflict.policy";
   public static final String DEFAULT_EXISTING_ENTITY_POLICY = ExistingEntityPolicy.ABORT.name();
+
+  public static final String MANAGED_DATA_POLICY_KEY =
+      HiveDatasetFinder.HIVE_DATASET_PREFIX + ".managed.data.conflict.policy";
```
Comment (ibuenros) [inline gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/HiveCopyEntityHelper.java]: Should this be `unmanaged` instead?

File: gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/HiveCopyEntityHelper.java [hunk 2]
```diff
@@ -755,14 +773,26 @@ protected static DiffPathSet fullPathDiff(HiveLocationDescriptor sourceLocation,
...
+          partition.isPresent() ? partition.get().getCompleteName() : helper.getDataset().getTable().getCompleteName(),
           Arrays.toString(desiredTargetExistingPaths.keySet().toArray())));
     }
+    // Unless, the policy requires us to delete such un-managed files - in which case: we will add the leftover files
+    // to the deletion list.
+    else if (desiredTargetExistingPaths.size() > 0) {
+      log.warn("Un-managed files detected in target file system, however deleting them because of the policy: " +
+        ManagedDataPolicy.DELETE_UNMANAGED_DATA);
+      for (Path delete : desiredTargetExistingPaths.keySet()) {
+        log.warn("Un-managed file to be deleted: " + delete);
```
Comment (ibuenros) [inline gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/HiveCopyEntityHelper.java]: I would prefer not to log every file in a separate line. Can we do a single line with all files to be deleted?

File: gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/HiveCopyEntityHelper.java (hunk 1 above)
Comment (abti) [inline gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/HiveCopyEntityHelper.java]: Yes, unmanaged would be more crisp, will change it.

**Pull request merged**